sdks/java/core/src/main/java/org/apache/beam/sdk/schemas/FromRowUsingCreator.java

```diff
@@ -126,21 +127,25 @@ public T apply(Row row) {
               valueType,
               typeFactory);
     } else {
-      if (type.getTypeName().isLogicalType()
-          && OneOfType.IDENTIFIER.equals(type.getLogicalType().getIdentifier())) {
+      if (type.isLogicalType(OneOfType.IDENTIFIER)) {
```

Comment:
Probably I need to look into the code more. Why is `OneOfType` is a special case, while there is a generic branch for other logical types?

Reply:
because the actual conversion type changes depending on the oneOf case type. there might be a better way to handle this, but I'm not sure what it is yet.